File: booking-service/src/main/java/it/roadies/booking_service/entities/Booking.java
package it.roadies.booking_service.entities;

import it.roadies.booking_service.entities.enumeration.BookingStatus;
import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "BOOKING")
@Data
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "user_id" , nullable = false)
    private String userId;

    @Column(name = "travel_id", nullable = false)
    private UUID travelId;

    @Enumerated(EnumType.STRING)
    @Column(name = "booking_status", nullable = false)
    private BookingStatus status;

    @Column(name = "total_price", nullable = false)
    private BigDecimal totalPrice;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "expires_at", nullable = false)
    private LocalDateTime expiresAt;

    @Column(name = "people_count", nullable = false)
    private Integer peopleCount;

    @Version
    private Long version;

    @OneToMany(mappedBy = "booking", fetch = FetchType.LAZY)
    private List<BookingMember> members;

    @PrePersist
    public void setCreatedAtAndStatus() {
        this.createdAt = LocalDateTime.now();
        if (this.status == null) {
            this.status = BookingStatus.PENDING;
        }
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: android-app/app/src/main/java/it/roadies/android_app/model/Booking.kt
package it.roadies.android_app.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import it.roadies.android_app.client.models.booking.DepartureType
import java.math.BigDecimal
import java.time.LocalDateTime
import java.util.UUID

@Entity(tableName = "booking")
class Booking(@PrimaryKey val id: UUID,
              @ColumnInfo(name = "principal_id") val principalId: UUID?,
              @ColumnInfo(name = "total_price") val totalPrice: BigDecimal,
              @ColumnInfo(name = "people_count") val peopleCount: Int,
              @ColumnInfo(name = "start_date") val startDate: LocalDateTime,
              @ColumnInfo(name = "end_date") val endDate: LocalDateTime,
              @ColumnInfo(name = "title") val title: String,
              @ColumnInfo(name = "departure_type") val departureType: DepartureType?
)

